### PR TITLE
Refactor Project modules

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase/model/project/roi.ex [hunk 1]
```diff
@@ -0,0 +1,123 @@
+defmodule Sanbase.Model.Project.Roi do
```
Comment (tspenov) [inline lib/sanbase/model/project/roi.ex]: maybe `RoI` ?

Comment (IvanIvanoff) [reply lib/sanbase/model/project/roi.ex]: looks really unintuitive to me to have upper-lower-upper module name

File: lib/sanbase/model/project/project.ex [hunk 2]
```diff
@@ -315,89 +149,33 @@ defmodule Sanbase.Model.Project do
...
   end
 
-  def by_slug(slugs) when is_list(slugs) do
+  def by_slug(slugs, opts) when is_list(slugs) do
+    additional_preloads = Keyword.get(opts, :additional_preloads, [])
+
     Project
     |> where([p], p.coinmarketcap_id in ^slugs)
-    |> preload(^@preloads)
+    |> preload(^(@preloads ++ additional_preloads))
```
Comment (tspenov) [inline lib/sanbase/model/project/project.ex]: maybe wrap in func:
```
def preload_query(query, opts) do
  query
  |> preload(^(@preloads ++ additional_preloads))
end
```
so we can  remove `additional_preloads = Keyword.get(opts, :additional_preloads, [])` that line from all queries

File: lib/sanbase_web/graphql/resolvers/project/project_resolver.ex [hunk 3]
```diff
@@ -508,6 +504,25 @@ defmodule SanbaseWeb.Graphql.Resolvers.ProjectResolver do
     Cache.func(fn -> fetch_posts_by_ticker(ticker) end, {:related_posts, ticker}).()
   end
 
+  def available_metrics_for_project(_root, %{slug: slug}, _resolution) do
+    with %Project{} = project <- Project.by_slug(slug) do
+      is_erc20? = Project.is_erc20?(project)
+
+      case project do
+        %Project{coinmarketcap_id: "bitcoin"} ->
+          []
```
[the remark this answers is not in the document]
Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/resolvers/project/project_resolver.ex]: I'll remove it

File: lib/sanbase/model/project/project.ex (hunk 2 above)
Comment (IvanIvanoff) [reply lib/sanbase/model/project/project.ex]: done

**Comment by tspenov** [comment]
@IvanIvanoff build fails

**Pull request merged**